openedx/core/djangoapps/user_api/accounts/api.py

```diff
@@ -533,3 +534,34 @@ def _validate_email(email):
         raise AccountEmailInvalid(
             u"Email '{email}' format is not valid".format(email=email)
         )
+
+
+@intercept_errors(UserAPIInternalError, ignore_errors=[UserAPIRequestError])
+@transaction.atomic
+def delete_user(user):
```

Comment:
@john2x, isn't there an existing method (added for other deletion tickets) to delete a user that this code can be added to?

Reply:
@lgp171188 the one I'm aware of was the one for api-integration (introduced in this PR https://github.com/edx-solutions/api-integration/pull/188/). I created a separate PR for api-integration to use this new method https://github.com/edx-solutions/api-integration/pull/197 (sorry, should've included this in the description above, I'll update)

Reply:
@lgp171188 ah there is a conflict for this part https://github.com/edx-solutions/api-integration/blob/master/edx_solutions_api_integration/users/views.py#L395-L396. It looks like that line could be moved into this new `delete_user` method. Wdyt?

Reply:
I have resolved the conflict from api-integration and moved the comment retirement here.

Reply:
@viadanna, do you have any idea if there are any other existing methods (created for other stories in the epic) for adding the certificate deletion adding a new method for deleting user?

Reply:
I don't follow you question, @lgp171188 

Hawthorn should have the certificate deletion as part of the user retirement feature. The user deletion is just implemented here, as Hawthorn just anonymize the records. It seems we have two implementations, one in https://github.com/edx-solutions/api-integration/pull/188 and another here.

Reply:
@viadanna, sorry if I wasn't clear. Since this code aims to be the single entrypoint in LMS to delete users, I was wondering if all the deletion-related features should use this or if they have their own methods in LMS for the deletion.

Reply:
Currently we have `api-integration` grouping calls to delete edx user data and discussion data, but I think for profile images it makes sense keeping it here.

Reply:
@viadanna there's also a related PR for `api-integration` which changes the DELETE api there to use the `delete_user` method introduced here https://github.com/edx-solutions/api-integration/pull/197/files#diff-6e10f5856db69f12fe0c2aef7e4ec7dfR396